app/views/courses/moss.html.erb

```diff
@@ -1,177 +1,168 @@
+		</li>
 
-<h2>Run the Moss Cheat Checker</h2>
+	</ul>
+<% else %>
+	<% content_for :javascripts do %>
+		<script type="text/javascript">
+			/**
+			 * toggles the "style.display" of the element with the given id.
+			 **/
+			function toggleOptions(id) {
+				var show = document.getElementById(id);
+				show.style.display = (show.style.display == "none")? "block" : "none";
+			}
 
-<h4>Step 1:</h4>
-<p>Click on course names to see their assessments, then check box all of the assessments you want Moss to compare against. </br> If a course doesn't appear, ensure you are an instructor.</p>
+		</script>
+	<% end %>
 
-<h5 style="display:inline">Filter Courses (Keywords Separated by Space): </h5>
-<div class="input-field">
-	<input placeholder="Course Name" type="text" onkeyup="filterCourses(this.value.toLowerCase())" style="margin-bottom: 10px;">
-</div>
-<%= form_tag([:run_moss, @course], multipart: true) do %>
-  <ul class="moss-list">
-  <% for course in @courses %>
-    <li class="filterableCourse" id="<%= course.display_name %>">
-      <h5 onclick="toggleOptions('<%= course.id %>Assessments')"> 
-        <%= course.full_name %>
-      </h5>
-      <ul id="<%= course.id %>Assessments" class="moss-inner-list" style="display:none" >
-      <% for a in course.assessments do %>
-		<li>
-		  <label>
-          	<%= check_box_tag "assessments[#{a.id}]","1",false, :class => "filled-in" %> 
-          	<span> <%= a.display_name %> </span>
-		  </label> 
-		  <br>
-          <div id='<%= a.id %>OptionsDiv' style="padding-left:10px;" >
-			<!-- <b>Is Archive:</b> -->
-			<label>
-				<%= check_box_tag "isArchive[#{a.id}]", :class => "filled-in" %>
-				<span>Is Archive: </span> 
-			</label>
-			</br>
-            <i>Check this if the submissions will need to be extracted. (The handin filename was: <%= a.handin_filename %>)</i>
-            <br>
-            <b>Files to send to Moss:</b>
-						<%= text_field_tag "files[#{a.id}]", "*hello.c *.c", required: true %>
-            <i>This can be file names (foo.c) or patterns(*.c), space-separated</i>
-            <br>
-          </div>
-        </li>
-      <% end %>
-      </ul>
-    </li>
-  <% end %>
-  </ul>
+	<h4>Step 1:</h4>
+	<p>Check the box of all the assessments you want to send to Moss for cheat checking.</p>
 
-  <br>
+	<%= form_tag([:run_moss, @course], multipart: true) do %>
+		<ul class="moss-list">
+			<li class="filterableCourse" id="<%= @course.display_name %>">
+				<h5 onclick="toggleOptions('<%= @course.id %>Assessments')">
+					<%= @course.full_name %>
+				</h5>
+				<ul id="<%= @course.id %>Assessments" class="moss-inner-list">
+					<% for a in @course.assessments do %>
+						<li>
+							<label>
+								<%= check_box_tag "assessments[#{a.id}]","1",false, :class => "filled-in" %>
+								<span> <%= a.display_name %> </span>
+							</label>
+							<br>
+							<div id='<%= a.id %>OptionsDiv' style="padding-left:10px;" >
```

Comment:
Looks like you accidentally undid your removal of the `a.id`s.

I would suggest removing all four occurrences of `id='<%= a.id %>OptionsDiv'` (instead of just the `a.id`) -- it doesn't make sense to have multiple elements with the same `id`